pillarbox-cast/src/main/java/ch/srgssr/pillarbox/cast/PillarboxCastPlayerBuilder.kt

```diff
@@ -0,0 +1,144 @@
+     */
+    fun seekForwardIncrement(seekForwardIncrement: Duration) {
+        check(seekForwardIncrement > Duration.ZERO)
+        this.seekForwardIncrement = seekForwardIncrement
+    }
+
+    /**
+     * @param maxSeekToPreviousPosition The maximum position for which [PillarboxCastPlayer.seekToPrevious] seeks to the previous [MediaItem].
+     */
+    fun maxSeekToPreviousPosition(maxSeekToPreviousPosition: Duration) {
+        check(maxSeekToPreviousPosition >= Duration.ZERO)
+        this.maxSeekToPreviousPosition = maxSeekToPreviousPosition
+    }
+
+    /**
+     * Track selector
+     *
+     * @param trackSelector The [CastTrackSelector] to use.
+     */
+    fun trackSelector(trackSelector: CastTrackSelector) {
+        this.trackSelector = trackSelector
+    }
+
+    /**
+     * On cast session available
+     *
+     * @param onCastSessionAvailable The method to call when [SessionAvailabilityListener.onCastSessionAvailable].
+     * @receiver
+     */
+    fun onCastSessionAvailable(onCastSessionAvailable: PillarboxCastPlayer.() -> Unit) {
+        this.onCastSessionAvailable = onCastSessionAvailable
+    }
+
+    /**
+     * On cast session unavailable
+     *
+     * @param onCastSessionUnavailable The method to call when [SessionAvailabilityListener.onCastSessionUnavailable]
+     * @receiver
+     */
+    fun onCastSessionUnavailable(onCastSessionUnavailable: PillarboxCastPlayer.() -> Unit) {
+        this.onCastSessionUnavailable = onCastSessionUnavailable
+    }
+
+    /**
+     * Media item converter
+     *
+     * @param mediaItemConverter the [MediaItemConverter] to use.
+     */
+    fun mediaItemConverter(mediaItemConverter: MediaItemConverter) {
+        this.mediaItemConverter = mediaItemConverter
+    }
+
+    internal fun create(context: Context): PillarboxCastPlayer {
+        return PillarboxCastPlayer(
+            context.getCastContext(),
+            context,
+            mediaItemConverter,
+            seekBackIncrement.inWholeMilliseconds,
+            seekForwardIncrement.inWholeMilliseconds,
+            maxSeekToPreviousPosition.inWholeMilliseconds,
+            trackSelector,
+        ).apply {
+            if (onCastSessionAvailable == null && onCastSessionUnavailable == null) return@apply
+            setSessionAvailabilityListener(object : SessionAvailabilityListener {
+                override fun onCastSessionAvailable() {
+                    onCastSessionAvailable?.invoke(this@apply)
+                }
+
+                override fun onCastSessionUnavailable() {
+                    onCastSessionUnavailable?.invoke(this@apply)
+                }
+            })
+        }
+    }
+}
+
+/**
+ * Defines a factory for creating instances of [PillarboxCastPlayerBuilder].
+ *
+ * @param Builder The type of [PillarboxCastPlayerBuilder] that this factory creates.
+ */
+interface CastPlayerConfig<Builder : PillarboxCastPlayerBuilder> {
```

Comment:
Why not use `PlayerConfig<PillarboxCastPlayerBuilder>`?

Reply:
It would not cause naming issue when in the same file we have local and cast player?

Reply:
Actually, it won't work, because `PlayerConfig` takes a `PillarboxBuilder`, but `PillarboxCastPlayerBuilder` doesn't extend that class.